lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java

```diff
@@ -779,19 +780,80 @@ private void markWorkbasketForDeletion(String workbasketId)
             WorkbasketImpl workbasket = workbasketMapper.findById(workbasketId);
             workbasket.setMarkedForDeletion(true);
             workbasketMapper.update(workbasket);
-            distributionTargetMapper.deleteAllDistributionTargetsBySourceId(workbasketId);
-            distributionTargetMapper.deleteAllDistributionTargetsByTargetId(workbasketId);
-            workbasketAccessMapper.deleteAllAccessItemsForWorkbasketId(workbasketId);
+            deleteWorkbasketTablesReferences(workbasketId);
         } finally {
             taskanaEngine.returnConnection();
             LOGGER.debug("exit from markWorkbasketForDeletion(workbasketId = {}).", workbasketId);
         }
     }
 
+    private void deleteWorkbasketTablesReferences(String workbasketId) {
+        // delete workbasket and sub-tables
+        distributionTargetMapper.deleteAllDistributionTargetsBySourceId(workbasketId);
+        distributionTargetMapper.deleteAllDistributionTargetsByTargetId(workbasketId);
+        workbasketAccessMapper.deleteAllAccessItemsForWorkbasketId(workbasketId);
+    }
+
+    public BulkOperationResults<String, TaskanaException> deleteWorkbaskets(List<String> workbasketsIds)
+        throws NotAuthorizedException, InvalidArgumentException, WorkbasketInUseException, WorkbasketNotFoundException {
+        LOGGER.debug("entry to deleteWorkbaskets(workbasketId = {})", LoggerUtils.listToString(workbasketsIds));
+
+        taskanaEngine.checkRoleMembership(TaskanaRole.BUSINESS_ADMIN, TaskanaRole.ADMIN);
+
+        try {
+            taskanaEngine.openConnection();
+            if (workbasketsIds == null || workbasketsIds.isEmpty()) {
+                throw new InvalidArgumentException("List of WorkbasketIds must not be null.");
+            }
+
+            List<String> existingWorkbasketIds = workbasketMapper.findExistingWorkbaskets(workbasketsIds);
+            BulkOperationResults<String, TaskanaException> bulkLog = cleanNonExistingWorkbasketExists(
```

Comment:
I don't understand what this is good for. Just to verify that the workbasket (id) exists? Since we run into deleteWorkbasket later on, this check will be done implicitly, won't it? 
And I would expect to have the errors/exceptions from deleteWorkbasket in the bulklog. Is there a reason for doing it this way?

Reply:
Ok, instead of returning a bulklog with non existing workbaskets, I will filter the list excludingWorkbasketWithPendingTasks which I think it is more usefull.